src/transforms/aws_cloudwatch_logs_subscription_parser.rs

```diff
@@ -0,0 +1,259 @@
+use super::Transform;
+use crate::{
+    config::{log_schema, DataType, TransformConfig, TransformContext, TransformDescription},
+    event::Event,
+    internal_events::{
+        AwsCloudwatchLogsSubscriptionParserEventProcessed,
+        AwsCloudwatchLogsSubscriptionParserFailedParse,
+    },
+};
+use chrono::{serde::ts_milliseconds, DateTime, Utc};
+use serde::{Deserialize, Serialize};
+use std::iter;
+use string_cache::DefaultAtom as Atom;
+
+#[derive(Deserialize, Serialize, Debug, Clone, Derivative)]
+#[serde(deny_unknown_fields, default)]
+#[derivative(Default)]
+pub struct AwsCloudwatchLogsSubscriptionParserConfig {
+    pub field: Option<Atom>,
```

Comment:
I was believing that we're moving away from using `Atom`, might be good to avoid introducing it in new components if that's the case.

Reply:
Looking at the PR for that it looks like `Atom` is still being used for component configuration, just not internally in the `Event`. Example: https://github.com/timberio/vector/pull/4255/files#diff-25a17f1694d8da276ec6ca5083fb832c . If that is the case, I guess this should stay as-is?